Makefile

```diff
@@ -68,23 +74,35 @@ install-pip-tools:
 	pip3 install -q "pip-tools>=6.4"
 
 install-docs-dependencies:
-	pip install -r requirements/docs.txt
+	pip install -r requirements/${PYV}/docs.txt
 
 freeze-deps:
+	make ensure-dep-folder
 	make install-pip-tools
-	pip-compile -o requirements/app.txt requirements/app.in
-	pip-compile -o requirements/test.txt requirements/test.in
-	pip-compile -o requirements/dev.txt requirements/dev.in
-	pip-compile -o requirements/docs.txt requirements/docs.in
+	pip-compile -o requirements/${PYV}/app.txt requirements/app.in
+	# Create app_temp.txt to create constraints for test.txt and dev.txt
+	cat requirements/${PYV}/app.txt > requirements/app_temp.txt
+	pip-compile -o requirements/${PYV}/test.txt requirements/test.in
+	cat requirements/${PYV}/test.txt >> requirements/app_temp.txt
+	pip-compile -o requirements/${PYV}/dev.txt requirements/dev.in
+	cat requirements/${PYV}/app.txt > requirements/app_temp.txt
```

Comment:
This looks like we're overwriting all requirements we wrote earlier (using `>`, not `>>`).

I guess we should write three temp files, as in most .in files we are not interested in the superset of all requirements, but usually just app.txt (not also test.txt for instance)

Reply:
I wrote it in a way that since dev.txt depends on app.txt and test.txt, I will generate file with app.txt first and then copy the content of test.txt into it. Instead of having 2 separate files.

Reply:
I would opt for explicitly having one file each, use them one by one as we had before, then delete them all. Easier to understand.